cassiopeia/plotting/itol_utilities.py

```diff
@@ -0,0 +1,623 @@
+"""
+Basic utilities that can be used to generate visualizations of tree using the
+iTOL tree plotting interface. See: https://itol.embl.de/ for more information
+on the iTOL software and how to create an account.
+"""
+
+import os
+
+import configparser
+import shutil
+import tempfile
+from typing import Dict, List, Tuple, Optional
+
+from bokeh import palettes
+from ete3 import Tree
+from itolapi import Itol
+from itolapi import ItolExport
+from matplotlib.colors import hsv_to_rgb, rgb_to_hsv
+import numpy as np
+import pandas as pd
+from tqdm.auto import tqdm
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.preprocess import utilities
+
+
+class iTOLError(Exception):
+    """Raises errors related to iTOL plotting
+    """
+
+    pass
+
+
+def upload_and_export_itol(
```

Comment:
No, because I think some more advanced users (including myself) might want to have a bit more control over the plotting and thus it's nice to expose these other functions if the user imports the entire file.